Sources/RedisKit/RedisSet.swift

```diff
@@ -146,18 +161,17 @@
     ///     // returns 4 elements, with a duplicate
     ///     set.random(max: 4, allowDuplicates: true)
     ///
-    /// - Parameter max: The max number of elements to pull, as available.
-    /// - Parameter allowDuplicates: Should duplicate elements be picked?
-//    public func random(max: Int = 1, allowDuplicates: Bool = false) -> EventLoopFuture<[Element]?> {
-//        precondition(max > 0, "Max should be a positive value. Use 'allowDuplicates' to handle proper value sign")
-//
-//        guard max > 1 else { return random() }
-//
-//        let count = allowDuplicates ? -max : max
-//        return connectionPool.withConnection { $0.srandmember(self.id, max: count) }
-//            .map {
-//                guard let results = $0.array else { return nil }
-//                return try results.map { try Element.convertFromRESP($0) }
-//            }
-//    }
-//}
+    /// See `RedisClient.srandmember(from:max:)`
+    /// - Parameters:
+    ///     - max: The max number of elements to pull, as available.
+    ///     - allowDuplicates: Should duplicate elements be picked?
+    /// - Returns: The elements randomly selected from the set.
+    @inlinable
+    public func randomElements(max: Int = 1, allowDuplicates: Bool = false) -> EventLoopFuture<[Element]> {
+        assert(max > 0, "Max should be a positive value. Use 'allowDuplicates' to handle proper value sign")
```

Comment:
Should this be a `precondition` or just force `UInt`?

Reply:
Right now you are requiring that number to be `1` or greater, but if that can be changed to `0` or greater, than I am all for using a `UInt`.

Reply:
I was able to replace these with the following code:

```swift
let count: Int
if allowDuplicates {
    count = max >= Int.min.magnitude ? .min : Int(max)
} else {
    count = max >= Int.max ? .max : Int(max)
}
```

Reply:
I decided that I'm going to just leave this as an assert rather than do all the necessary math for handling UInt -> Int conversions (which could be error prone).

Users should easily be able to fix bugs triggered by the assert, and if they see the bugs in the wild there are enough hints in documentation, signature labels, and in the assert message to steer them in the direction of solving it.